### ISSUE TITLE
Trivy Rootfs reports only one PkgPath when vulnerable library is shipped in multiple jars    

### ISSUE THREAD

**Issue by rafaeltc** [aquasecurity/trivy#3297]
## Description

When using Trivy Rootfs to scan a lib directory with multiple jar (json-smart-2.3.jar, wiremock-standalone-2.27.2.jar), Trivy only reports the CVE-2021-27568 in either  json-smart-2.3.jar or wiremock-standalone-2.27.2.jar, but never both PkgPath simultaneously. 

When scanning the jars individually, Trivy identifies the same vulnerability (CVE-2021-27568) in each jar.

## What did you expect to happen?
When scanning the directory containing both `json-smart-2.3.jar` and `wiremock-standalone-2.27.2.jar` with `trivy rootfs -f json` I would expect the following entries:

```json
{
   "VulnerabilityID": "CVE-2021-27568",
   "PkgName": "net.minidev:json-smart",
   "PkgPath": "wiremock-standalone-2.27.2.jar",
   ...
}
{
   "VulnerabilityID": "CVE-2021-27568",
   "PkgName": "net.minidev:json-smart",
   "PkgPath": "json-smart-2.3.jar",
    ...
}
```

## What happened instead?

The output only reports the vulnerability on `PkgPath = json-smart-2.3.jar`:

```json
"VulnerabilityID": "CVE-2021-27568",
"PkgName": "net.minidev:json-smart",
"PkgPath": "json-smart-2.3.jar",
```

## Output of run with `-debug`:

```
DEBUG	Severities: ["HIGH" "CRITICAL"]
DEBUG	cache dir:  /Users/foobar/Library/Caches/trivy
DEBUG	DB update was skipped because the local DB is the latest
DEBUG	DB Schema: 2, UpdatedAt: 2022-12-13 12:07:14.884952254 +0000 UTC, NextUpdate: 2022-12-13 18:07:14.884951854 +0000 UTC, DownloadedAt: 2022-12-13 12:37:29.746565 +0000 UTC
INFO	Vulnerability scanning is enabled
DEBUG	Vulnerability type:  [os library]
INFO	Secret scanning is enabled
INFO	If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
INFO	Please see also https://aquasecurity.github.io/trivy/v0.35/docs/secret/scanning/#recommendation for faster secret detection
DEBUG	No secret config detected: trivy-secret.yaml
DEBUG	Walk the file tree rooted at '.' in parallel
DEBUG	Parsing Java artifacts...	{"file": "json-smart-2.3.jar"}
DEBUG	Parsing Java artifacts...	{"file": "wiremock-standalone-2.27.2.jar"}
DEBUG	OS is not detected.
DEBUG	Detected OS: unknown
INFO	Number of language-specific files: 1
INFO	Detecting jar vulnerabilities...
DEBUG	Detecting library vulnerabilities, type: jar, path:
```

## Output of `trivy -v`:

```
Version: 0.35.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-12-13 12:07:14.884952254 +0000 UTC
  NextUpdate: 2022-12-13 18:07:14.884951854 +0000 UTC
  DownloadedAt: 2022-12-13 12:37:29.746565 +0000 UTC
```

### Hints 

Caching seems to affect the output since the output of scanning the whole dir will change if `wiremock-standalone-2.27.2.jar` was scanned beforehand:

- If only `wiremock-standalone-2.27.2.jar` is scanned before scanning the whole dir, then the output includes:

```
   "VulnerabilityID": "CVE-2021-27568",
   "PkgName": "net.minidev:json-smart",
   "PkgPath": "wiremock-standalone-2.27.2.jar"
``` 

- otherwise, the output includes instead:

```
"VulnerabilityID": "CVE-2021-27568",
"PkgName": "net.minidev:json-smart",
"PkgPath": "json-smart-2.3.jar"
```

**Note** that this behavior only occurs if the cache is **not** cleared between scanning `wiremock-standalone` and the whole dir.

### Outputs
- [trivy_output_full.txt](https://github.com/aquasecurity/trivy/files/10219706/trivy_output_full.txt): `trivy rootfs -f json --debug -o trivy_output_full.txt  --severity "HIGH,CRITICAL" .`
- [trivy_output_jsonsmart.txt](https://github.com/aquasecurity/trivy/files/10219707/trivy_output_jsonsmart.txt): `trivy rootfs -f json --debug -o trivy_output_jsonsmart.txt  --severity "HIGH,CRITICAL" ./json-smart-2.3.jar` 
- [trivy_output_wiremock.txt](https://github.com/aquasecurity/trivy/files/10219708/trivy_output_wiremock.txt): `trivy rootfs -f json --debug -o trivy_output_wiremock.txt  --severity "HIGH,CRITICAL" ./wiremock-standalone-2.27.2.jar` 

### SHA256
00b14f9d6399a9bba96139f84354f981d9fd1c3f4cf6938d36391f292f76382b  wiremock-standalone-2.27.2.jar
903f48c8aa4c3f6426440b8d32de89fa1dc23b1169abde25e4e1d068aa67708b  json-smart-2.3.jar

**Comment by DmitriyLewen** [comment]
Hello @rafaeltc 
Thanks for your report!

Looks like it is duplicate of #3274 
I close this issue. Fill free to reopen this issue, if i am wrong.

Regards, Dmitiry
